docs: reorganize docker-compose docs, clean up admin homepage (#23029)
sourcegraph/sourcegraph: doc/admin/install/docker-compose/operations.md

# Operations guides for Sourcegraph with Docker Compose

Operations guides specific to managing [Sourcegraph with Docker Compose](./index.md) installations.

> 

Reviewer: ```suggestion

If you are not currently running sourcegraph via Docker Compose and would like to migrate, please see the [migration](./migrate.md).
```

Reviewer: Moved from lower where it felt out of place among upgrade info.

Author: Hmm I moved it from the top because it seems like a very rare thing to need - do we have that many customers upgrading from single-container to warrant featuring it so prominently? I think it's fine to leave it tucked away a bit and available primarily via search

## Featured guides

<div class="getting-started">
  <a href="#configure" class="btn btn-primary" alt="Configure">
   <span>Configure</span>
   </br>
   Configure your Sourcegraph deployment with our deployment reference.
  </a>

  <a href="#upgrade" class="btn" alt="Upgrade">
   <span>Upgrade</span>
   </br>
   Upgrade your deployment to the latest Sourcegraph release.
  </a>

  <a href="#backup-and-restore" class="btn" alt="Backup and restore">
   <span>Backup and restore</span>
   </br>
   Back up your Sourcegraph instance and restore from a previous backup.
  </a>
</div>

## Configure

> We **strongly** recommend that you create your own fork of [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) to track customizations to the [Sourcegraph Docker Compose yaml](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/docker-compose.yaml). **This will make upgrades far easier.**

Reviewer: ```suggestion
We **strongly** recommend that you create and run Sourcegraph from your own fork of [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) to track customizations to the [Sourcegraph Docker Compose yaml](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/docker-compose.yaml). **This will make upgrades far easier.**
```

> - Fork the [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) repository and clone your fork

Reviewer: ```suggestion
- Fork the [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) repository and clone your fork
    - [How to fork a repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository)
    - Clone the fork using `git clone <repo-fork-url>`
```

  
  > WARNING: Set it to **private** if you plan to store secrets (SSL certificates, external Postgres credentials, etc.) within the repository.

- Add the upstream remote so that you can [get updates](#upgrade).

```bash
git remote add upstream https://github.com/sourcegraph/deploy-sourcegraph-docker
```

- Create a `release` branch to track all of your customizations to Sourcegraph. When you [upgrade Sourcegraph](#upgrade), you will merge upstream into this branch.

```bash

> export SOURCEGRAPH_VERSION="v3.26.3"

Reviewer: ```suggestion
export SOURCEGRAPH_VERSION="v3.26.3" # Specify the version you want to install
```

Author: de2c4b8886a4a37287ed98ddc103d497424c9d48

> git checkout $SOURCEGRAPH_VERSION -b release

Reviewer: ```suggestion
git checkout $SOURCEGRAPH_VERSION -b release # Check out the selected version for use, in a new branch called "release"
```

Author: 4e67a833bc69ceeb0ed37031e69c73decd5848eb

```

- Commit customizations to the [Sourcegraph Docker Compose yaml](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/docker-compose.yaml) to your `release` branch

## Upgrade

This requires you to have [set up configuration for Docker Compose](#configure).

When you upgrade, merge the corresponding upstream release tag into your `release` branch.

```bash
# fetch updates
git fetch upstream
# to merge the upstream release tag into your release branch.
git checkout release
git merge upstream v$SOURCEGRAPH_VERSION
```

Address any merge conflicts you might have.

> If you are upgrading a live deployment, make sure to check the [release upgrade notes](../../updates/docker_compose.md) for any additional actions you need to take. Then run:

Reviewer: ```suggestion
If you are upgrading a live deployment, make sure to check the [release upgrade notes](../../updates/docker_compose.md) for any additional actions you need to take **before proceeding**. Then run:
```

```bash

> docker-compose down --remove-orphans

Reviewer: ```suggestion
docker-compose down --remove-orphans # Fully stop the Docker Compose instance of Sourcegraph currently running
```

> docker-compose up -d

Reviewer: ```suggestion
docker-compose up -d # Start Docker Compose again, now using the latest contents of the branch you're in
```

## Manage storage

The [Sourcegraph Docker Compose definition](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/docker-compose.yaml) uses [Docker volumes](https://docs.docker.com/storage/volumes/) to store its data. These volumes are stored at `/var/lib/docker/volumes` by [default on Linux](https://docs.docker.com/storage/#choose-the-right-type-of-mount).

Guides for managing cloud storage and backups are available in our [cloud-specific installation guides](./index.md#cloud-installation):

- [Storage and backups for Amazon Web Services](./aws.md#storage-and-backups)
- [Storage and backups for Google Cloud](./google_cloud.md#storage-and-backups)
- [Storage and backups for Digital Ocean](./digitalocean.md#storage-and-backups)

## Backup and restore

The following instructions are specific to backing up and restoring the sourcegraph databases in a Docker Compose deployment. These do not apply to other deployment types.

> If you are not currently running sourcegraph via Docker Compose and would like to migrate, please see the [migration](./migrate.md).

Reviewer: ```suggestion
```

### Only core data will be backed up

These instructions will only back up core data including user accounts, configuration, repository-metadata, etc. Other data will be regenerated automatically:

* Repositories will be re-cloned
* Search indexes will be rebuilt from scratch

> The above may take awhile if you have a lot of repositories. In the meantime, searches may be slow or return incomplete results. Usually this process will not take longer than 6 hours.

Reviewer: ```suggestion
The above may take a while if you have a lot of repositories. In the meantime, searches may be slow or return incomplete results. This process rarely takes longer than 6 hours and is usually **much** faster.
```
